ibis/expr/types/relations.py

```diff
@@ -88,6 +91,81 @@ def _repr_html_(self) -> str | None:
 
         return self.execute()._repr_html_()
 
+    def __rich_console__(self, console, options):
```

Comment:
It there another way to add rich support for user types? If not could we move parts of the code to `ibis.common.pretty`?

These methods are fairly large and rather noisy.

Reply:
This is the way to do it. I can move the code.

Reply:
Consolidated!

Reply:
Much better, thanks!